### ISSUE TITLE
Lookahead for XML files

### ISSUE THREAD

**Issue by mr-martian** [apertium/apertium-recursive#8]
Using `trx-comp` we can compile files in the `.t*x` format and use them as recursive rules. However, `trx-comp` does not generate lookahead paths and so loses the speedups from #6.

Doing this will require working out the part of speech tag of every generated chunk, which may force all chunks to have literal pos tags.

**Comment by mr-martian** [comment]
One possible solution would be to slightly modify the xml format.
```xml
<rule firstChunk="NP">
```
There should probably also be a way of specifying that the output could be one of a few things:
```xml
<rule firstChunk="S DP"> <!-- sentence or relative clause ? -->
```
Then lookahead could be constructed from that attribute and the `<pattern>` elements.

@mlforcada

**Comment by mr-martian** [comment]
Implemented in de1a03afd3456edd3328b0a8a3b683390a0b17c9
